### PR TITLE
Simplify check for neighboring elements in Minmod

### PR DESCRIPTION
**Pull request by fmahebert**
## Proposed changes

A tiny change to help @nilsdeppe's boundary changes.

### Types of changes:

- [ ] Bugfix
- [ ] New feature
- [ ] Refactor

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: src/Evolution/DiscontinuousGalerkin/Limiters/MinmodHelpers.cpp [hunk 1]
```diff
@@ -73,9 +73,8 @@ double effective_difference_to_neighbor(
     const Side& side,
     const DirectionMap<VolumeDim, double>& effective_neighbor_means,
     const DirectionMap<VolumeDim, double>& effective_neighbor_sizes) noexcept {
-  const auto& externals = element.external_boundaries();
   const auto dir = Direction<VolumeDim>(dim, side);
-  const bool has_neighbors = (externals.find(dir) == externals.end());
+  const bool has_neighbors = element.neighbors().contains(dir);
```
Comment (nilsvu) [inline src/Evolution/DiscontinuousGalerkin/Limiters/MinmodHelpers.cpp:77]: How does this change help with boundary conditions, are you planning to make changes to `element.external_boundaries()` or `element.neighbors()`?

Comment (fmahebert) [reply src/Evolution/DiscontinuousGalerkin/Limiters/MinmodHelpers.cpp:77]: @nilsdeppe is thinking of changing/removing `element.external_boundaries()`. The code change here means one less place to worry about side effects... and is also cleaner and easier to understand, so a good improvement regardless.

Comment (nilsdeppe) [reply src/Evolution/DiscontinuousGalerkin/Limiters/MinmodHelpers.cpp:77]: just getting the external boundaries without their boundary conditions isn't a useful operation. Or, at least, we don't do it anywhere in the code and I don't see why you would need it. This means I want to remove `element.external_boundaries` and replace `block.external_boundaries()` with `block.boundary_conditions()`.

**Review by nilsdeppe — APPROVED** [review]
Thanks for doing this :)

**Review by nilsvu — APPROVED** [review]
Thanks for explaining. I'll have a look through my unmerged code to see how what you're suggesting affects it. But the changes in this PR are good regardless :)

**Review by kidder — APPROVED** [review]

**Pull request merged**